cmake/modules/RootMacros.cmake

```diff
@@ -1522,6 +1521,7 @@ function(ROOT_ADD_GTEST test_suite)
   # against. For example, tests in Core should link only against libCore. This could be tricky
   # to implement because some ROOT components create more than one library.
   ROOT_EXECUTABLE(${test_suite} ${source_files} LIBRARIES ${ARG_LIBRARIES})
+  target_include_directories(${test_suite} PUBLIC ${CMAKE_CURRENT_BINARY_DIR} ${GMOCK_INCLUDE_DIR} ${CMAKE_SOURCE_DIR}/test ${GTEST_INCLUDE_DIR})
```

Comment:
why we need `${CMAKE_SOURCE_DIR}/test`?
(check how we are using ROOT_ADD_GTEST: https://github.com/root-project/root/search?q=ROOT_ADD_GTEST&unscoped_q=ROOT_ADD_GTEST)

Reply:
I agree it is not the best we can do. However, we need this folder because we want to make `ROOTUnitTestSupport.h` accessible to all tests for inclusion. Do you have an opinion where this header should reside?

@Axel-Naumann, @pcanal, what about moving this new "core" testing infrastructure in some place in core, eg core/testing_support/ROOTUnitTestSupport.h?

Reply:
> However, we need this folder because we want to make ROOTUnitTestSupport.h accessible to all tests for inclusion.

I guess this can be solved with an interface library:
1. We create a target whose only job it is to set up the include for `ROOTUnitTestSupport.h`:
   `add_library(ROOTUnitTestSupport INTERFACE)`
2. We set the include directory that any user of this library should inherit:
   `target_include_directories(ROOTUnitTestSupport INTERFACE ${CMAKE_SOURCE_DIR}/test)`
   (Or wherever this header ends up. The beauty is that you only have to get it right here.)
3. Now, all tests that need this header are defined as:
```
ROOT_ADD_GTEST(testWithRUTSupp myTest.cxx
                                 LIBRARIES lib1 lib2 ROOTUnitTestSupport)
```
Like this, it would not pollute the includes of all other gtests in ROOT. Further, you can move around the header, since you only need to update the location in 2. When you realise that at some point header-only is not enough, you just add stuff to the target in step 2. and all the gtests that use it will just follow.

Reply:
And yes, if you put it in `core/testing_support`, a two-line CMakeLists.txt with the above commands should get it right.

Reply:
I hope that after https://github.com/root-project/root/pull/5228 all problems will go away :)

Reply:
@hageboeck, I find it awkward to spell `LIBRARIES Core Hist ROOTUnitTestSupport`, i.e. it seems that `ROOTUnitTestSupport` should not go in this bucket. What about enabling it in the `ROOT_ADD_GTEST`? Even if we make it a library one day (I do not expect that to happen anytime soon) it will be a very small one which won't contribute a lot to the overall binary size. On the pro side -- we will have available and ready to use.

Reply:
@vgvassilev What do you mean by "enabling" it? Do you mean something like
```
ROOT_ADD_GTEST(myTest myTest.cxx
   LIBRARIES A B
   ENABLE_UNIT_TEST_SUPPORT)
```
And then inside ROOT_ADD_GTEST do the necessary setup?

It can be done. It's just that `INTERFACE` libraries in CMake will get everything right if you use them correctly. So I would recommend to use that, but *where* you "link", that is, pull in the interface of this target, doesn't matter.

Reply:
@hageboeck, I meant `ROOT_ADD_GTEST(myTest myTest.cxx LIBRARIES A B)` and inside `ROOT_ADD_GTEST` we call `ENABLE_UNIT_TEST_SUPPORT`. That would turn the feature on for every gtest we have.

I do not mind using a INTERFACE library underneath.